File: challenging/9_delayed_output/delayed_output.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func sleep(ms int) {
	time.Sleep(time.Duration(ms) * time.Millisecond)
}

func log(s string, delay int, randomized bool) {
	for _, c := range s {
		fmt.Printf("%c", c)
		if randomized {
			sleep(rand.Intn(delay))
		} else {
			sleep(delay)
		}
	}
	fmt.Printf("\n")
}

func main() {
	log("Hello, world!", 200, true)
}

File: challenging/9_delayed_output/delayed_output.js
const sleep = (ms) => {
  return new Promise(resolve => setTimeout(resolve, ms));
};

const log = async (s, delay, randomized) => {
  for (const c of s) {
    process.stdout.write(c);
    await sleep((randomized ? Math.random() : 1) * delay);
  }
  process.stdout.write('\n');
}

log('Hello, world!', 200, true);
